### PR TITLE
FEATURE: Redesigned bookmark modal and menu

### PR DESCRIPTION
**Pull request by martin-brennan**
Adds the new quick menu for bookmarking. When you bookmark
a post (chat message behaviour will come later) we show this new quick
menu and bookmark the item straight away:

You can then choose a reminder quick option, or choose Custom... to open
the old modal. If you click on an existing bookmark, we show the same quick menu
but with Edit and Delete options:

A later PR will introduce a new bookmark modal, but for now we
are using the old modal for Edit and Custom... options.

### CODE REVIEW

**Comment by martin-brennan** [comment]
Just merged main into this so it's not too far out of date and used `DMenu` for the bookmark-menu just to get it working though looking crap. Will keep working on it bit by bit.

File: app/assets/javascripts/discourse/app/lib/post-bookmark-manager.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { setOwner } from "@ember/application";
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/lib/post-bookmark-manager.js:1]: We would also have a TopicBookmarkManager and ChatBookmarkManager and insert whatever one makes sense into BookmarkMenu

**Comment by martin-brennan** [comment]
@jjaffeux IMO if you approve and merge this it would be better, then we can split up and work on various parts of it and not end up with a mega-PR that is 8 months old. It's behind a feature flag, nothing will be impacted.

File: app/assets/javascripts/discourse/app/components/bookmark-menu.gjs [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+      this.onBookmark();
+    }
+  }
+
+  @action
+  onRegisterApi(api) {
+    if (!this.dMenu) {
+      this.dMenu = api;
+    }
+  }
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/bookmark-menu.gjs:101]:
```suggestion
  onRegisterApi(api) {
    this.dMenu = api;
  }
```

I don't see much trouble if we ever overwrite this, it should be the same

File: app/assets/stylesheets/common/components/bookmark-menu.scss [hunk 3]
```diff
@@ -0,0 +1,113 @@
+@keyframes slide {
+  0% {
+  }
```
Comment (jjaffeux) [inline app/assets/stylesheets/common/components/bookmark-menu.scss:3]: do you need this?

File: app/assets/stylesheets/common/components/bookmark-menu.scss [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  &__notice {
+    position: absolute;
+    top: -1.5rem;
+    left: 50%;
+    transform: translateX(-50%);
+    color: var(--tertiary);
+    padding: 0.25rem 0.5rem;
+    white-space: nowrap;
+    opacity: 1;
+
```
Comment (jjaffeux) [inline app/assets/stylesheets/common/components/bookmark-menu.scss:20]: probably nice to add: "will-change: auto;"

File: app/assets/stylesheets/common/components/bookmark-menu.scss [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    &__row {
+      border-bottom: 1px solid var(--primary-low);
+      width: 100%;
+      display: flex;
+      align-items: left;
+
+      &:hover,
+      &:focus {
+        background: var(--tertiary-very-low) !important;
```
Comment (jjaffeux) [inline app/assets/stylesheets/common/components/bookmark-menu.scss:54]: I'm surprised we need so many !important

**Review by jjaffeux — APPROVED** [review]
I think it's good enough to merge and iterate from here

File: app/assets/stylesheets/common/components/bookmark-menu.scss (hunk 3 above)
Comment (martin-brennan) [reply app/assets/stylesheets/common/components/bookmark-menu.scss:3]: Ah no this seems like it's from the old notice before we changed to toast, removing

File: app/assets/stylesheets/common/components/bookmark-menu.scss (hunk 4 above)
Comment (martin-brennan) [reply app/assets/stylesheets/common/components/bookmark-menu.scss:20]: Removed this anyway

File: app/assets/stylesheets/common/components/bookmark-menu.scss (hunk 5 above)
Comment (martin-brennan) [reply app/assets/stylesheets/common/components/bookmark-menu.scss:54]: I got rid of them, didn't seem to affect anything

**Pull request merged**